app/backend/core/message_truncater/message_shortener.py

```diff
@@ -0,0 +1,100 @@
+import logging
+import unicodedata
+from collections.abc import Iterable
+from typing import Optional, Union
+
+from openai.types.chat import (
+    ChatCompletionContentPartParam,
+    ChatCompletionMessageParam,
+    ChatCompletionNamedToolChoiceParam,
+    ChatCompletionSystemMessageParam,
+    ChatCompletionToolParam,
+    ChatCompletionUserMessageParam,
+)
+
+from .model_helper import count_tokens_for_message, count_tokens_for_system_and_tools
+
+
+# Normalises the content of a message to NFC form to ensure the consistency
+# between the messages format.
+def normalize_content(content: Union[str, Iterable[ChatCompletionContentPartParam]]):
```

Comment:
Let's make this a proper docstring (underneath the function signature, and with proper formatting).